client/my-sites/plans-features-main/hooks/use-action-callback.ts

```diff
@@ -0,0 +1,188 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import {
+	applyTestFiltersToPlansList,
+	PRODUCT_1GB_SPACE,
+	type PlanSlug,
+	isWpcomEnterpriseGridPlan,
+	isFreePlan,
+	getPlanPath,
+} from '@automattic/calypso-products';
+import page from '@automattic/calypso-router';
+import { AddOns, Plans } from '@automattic/data-stores';
+import { useCallback } from '@wordpress/element';
+import { getPlanCartItem } from 'calypso/lib/cart-values/cart-items';
+import { addQueryArgs } from 'calypso/lib/url';
+import { useFreeTrialPlanSlugs } from 'calypso/my-sites/plans-features-main/hooks/use-free-trial-plan-slugs';
+import useCurrentPlanManageHref from './use-current-plan-manage-href';
+import type {
+	UseActionCallbackParams,
+	PlansIntent,
+	UseActionCallback,
+} from '@automattic/plans-grid-next';
+import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
+
+function useUpgradeHandler(
+	siteSlug?: string | null,
+	withDiscount?: string,
+	cartHandler?: ( cartItems?: MinimalRequestCartProduct[] | null ) => void
```

Comment:
In general, I much prefer single object props (spread -> `{ prop1, prop2, etc. }`), rather than enumerating properties this way. It's easier to extend and maintain, order won't matter, and also more consistent with React environment.

Reply:
Explanation makes sense, especially about consistency with react environment. Making the change 👍

Reply:
[9989ba0](https://github.com/Automattic/wp-calypso/pull/88694/commits/9989ba040c427175bb136b39721e1acd998f0db6)